### DISCUSSION TITLE
install this module

### DISCUSSION THREAD

**Discussion by jxmes7** [Rapptz/discord.py#9731] [General]
where and how can i download and install  discord.py==2.4.0a4987+gbd402b48

**Comment by Kile** [comment]
It seems like you're trying to install a specific commit. Just with this reduced hash you can't tell which commit exactly, but I am assuming the main priority is to install new features from the `master` branch. For this, run `pip3 install git+https://github.com/Rapptz/discord.py.git`

**Comment by LostLuma** [comment]
For anyone else reading this, the metadata part of discord.py's alpha versions tells you exactly which commit was used to build the library at the time. The `g` tells you it's part of the git revision info, the rest of it (`bd402b48`) is the short git hash.

You can then install the library off a specific commit in the following way:
`pip install -U "git+https://github.com/Rapptz/discord.py@bd402b4"`
Or if you have the full hash:
`pip install -U "git+https://github.com/Rapptz/discord.py@bd402b486cc12f0c1bf7377fd65f2fe0a8fabd73"`
Alternatively there's also this syntax which allows installing optional dependency profiles:
`pip install -U "discord.py[speed] @ git+https://github.com/Rapptz/discord.py@bd402b486cc12f0c1bf7377fd65f2fe0a8fabd73"`
